policy-controller/core/src/http_route.rs

```diff
@@ -176,3 +197,51 @@ impl PartialEq for QueryParamMatch {
 }
 
 impl Eq for QueryParamMatch {}
+
+impl Status {
+    pub fn collect_from(status: Option<gateway::HttpRouteStatus>) -> Vec<Self> {
+        status
+            .into_iter()
+            .map(|status| {
+                status
+                    .inner
+                    .parents
+                    .iter()
+                    .filter_map(|parent_status| Self::from_parent_status(parent_status))
+                    .collect::<Vec<_>>()
+            })
+            .flatten()
+            .collect()
+    }
+
+    fn from_parent_status(status: &gateway::RouteParentStatus) -> Option<Self> {
+        if let Some(ref kind) = status.parent_ref.kind {
+            if kind != "Server" {
+                return None;
+            }
+        }
```

Comment:
this looks like it will return `Some` if the parent ref doesn't have a `kind`? is that correct?

Reply:
Yep good catch fixed in 54de3530d083a0a7af10717bc97b6bbf98340da0.